src/libexpr-tests/yaml.cc

```diff
@@ -0,0 +1,425 @@
+    EXPECT_EQ(1. / val.fpoint(), 1. / -0.) << "\"!!float -00\" shall be parsed as -0.0";
+
+    val = parseYAML("!!float 9223372036854775808"); // INT64_MAX + 1
+    ASSERT_EQ(val.type(), nFloat);
+    EXPECT_EQ(val.fpoint(), 9223372036854775808.) << "shall have no integer overflow";
+
+    val = parseYAML("!!float -9223372036854775809"); // INT64_MIN - 1
+    ASSERT_EQ(val.type(), nFloat);
+    EXPECT_EQ(val.fpoint(), -9223372036854775809.) << "shall have no integer overflow";
+
+    const char * strings[] = {"0x1.", "0X1.", "0b1.", "0B1.", "0o1.", "0O1"};
+    for (auto str : strings) {
+        Value val = parseYAML(str);
+        ASSERT_EQ(val.type(), nString) << "'" << str << "' shall not be converted to a float";
+        EXPECT_EQ(val.string_view(), str);
+    }
+}
+
+TEST_F(FromYAMLTest, Overflow)
+{
+    const char * strings[] = {
+        "1e310",   // NixFloat overflow
+        "-1e310",  // NixFloat overflow
+        "1e-400",  // NixFloat underflow
+        "-1e-400", // NixFloat underflow
+        "1e-310",  // denormal
+        "-1e-310", // denormal
+        "!!float 11111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111"
+        "111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111"
+        "11111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111", // NixFloat overflow
+        "0x8000000000000000",               // integer overflow (INT64_MAX + 1)
+        "!!float 0x8000000000000000",       // integer overflow (does not match !float regex)
+        "0o1000000000000000000000",         // integer overflow (INT64_MAX + 1)
+        "!!float 0o1000000000000000000000", // integer overflow (does not match !float regex)
+        "9223372036854775808",              // integer overflow (INT64_MAX + 1)
+        "-9223372036854775809"              // integer underflow (INT64_MIN - 1)
+    };
+    for (auto str : strings) {
+        EXPECT_THROW(parseYAML(str), EvalError) << str << ": shall not succeed due to denormal/overflow/underflow";
+    }
+}
+
+TEST_F(FromYAMLTest, TrueYAML1_2)
+{
+    Value val = parseYAML("[ true, True, TRUE ]");
+    for (auto item : val.listItems()) {
+        ASSERT_EQ(item->type(), nBool);
+        EXPECT_TRUE(item->boolean());
+    }
+    const char * strings[] = {"y", "Y", "on", "On", "ON", "yes", "Yes", "YES"};
+    for (auto str : strings) {
+        Value val = parseYAML(str);
+        ASSERT_EQ(val.type(), nString) << "'" << str << "' shall not be converted to a boolean";
+        EXPECT_EQ(val.string_view(), std::string_view(str));
+    }
+}
+
+TEST_F(FromYAMLTest, TrueYAML1_1)
+{
+    Value options;
+    auto bindings = state.buildBindings(1);
+    bindings.alloc("useBoolYAML1_1").mkBool(true);
+    options.mkAttrs(bindings);
+
+    Value val = parseYAML("[ true, True, TRUE, y, Y, on, On, ON, yes, Yes, YES ]", options);
```

Comment:
question: is this really yaml ( then I didnt know) or does Nix parse these as booleans?

Reply:
Sadly, `YAML 1.1` specifies that this should be parsed as booleans, e.g. [kubernetes](https://kubernetes.io/docs/concepts/configuration/overview/) relies on this behavior. By default Nix uses `YAML 1.2`, where this is parsed as strings.

Reply:
It's a misconception that YAML would be simple. Instead the standard is very complex and making `YAML 1.2` a superset of JSON didn't simplify it.

Reply:
Hmm. jeah thats why people probably want toml over yaml sometimes. I am happy when this gets merged, not sure whats taking so long, ok its a new dependency? I guess the ecosystem benefits from such a function a lot. 

Does that yaml parser also support references?